pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -1230,11 +1306,15 @@ public Reference getReference() throws NamingException {
 
   public void setFromReference(Reference ref) {
     databaseName = getReferenceProperty(ref, "databaseName");
-    String port = getReferenceProperty(ref, "portNumber");
-    if (port != null) {
-      portNumber = Integer.parseInt(port);
+    String ports = getReferenceProperty(ref, "portNumber");
+    if (ports != null) {
+      String[] ps = ports.split(",");
+      portNumbers = new int[ps.length];
+      for (int i = 0; i < ps.length; i++) {
+        portNumbers[i] = Integer.parseInt(ps[i]);
```

Comment:
This should delegate to `setPortNumbers(...)`.

Reply:
done